ddtrace/_trace/_span_pointer.py

```diff
@@ -54,3 +55,31 @@ def __init__(
     def __post_init__(self):
         # Do not want to do the trace_id and span_id checks that SpanLink does.
         pass
+
+
+class _StandardHashingRules:
```

Comment:
This is a bit strange, but we can probably remove that part of the comment. 

One thing I don't understand is why we can't just have a method that uses a global value set to `self.desired_bits // self.bits_per_hex_digit` for the calculation. 

Or do we need this as a class because `"allows us to write tests for the chosen base_hashing_function instead of having to add runtime sanity checks."` ? 
However isn't the `base_hashing_function` just hardcoded here to use sha256 so it won't change and we could just always use that value?

I'm not actually against the way this is written, I'd just like to understand better.

Reply:
an original version of this code said something like "assert num_bits_in_hash >= num_required_bits" but our tooling complained and it seemed silly to do the `desired_bits // bits_per_hex_digit` calculation and comparison every time the hashing function is called.

i wanted a way to encapsulate this all these "business rules" without having to use global variables or setup that hard-codes sha256 in both the production code and the tests.

now that i say that, if i loosen the constraint of "don't say `sha256` twice" i can write the `_standard_hashing_function` as a proper function and still write tests that validate that its output is the right size and such. will write that up tomorrow and see how that looks.

Reply:
done! i've rearranged the code to have a "simpler" though more obscure hashing function and more descriptive tests